File: android/src/main/kotlin/ch/waio/pro_video_editor/src/features/render/helpers/ApplyScale.kt
import android.util.Log
import androidx.media3.common.Effect
import androidx.media3.common.util.UnstableApi
import androidx.media3.effect.ScaleAndRotateTransformation

@UnstableApi
fun applyScale(
    videoEffects: MutableList<Effect>, scaleX: Float?, scaleY: Float?,
) {
    if (scaleX == null && scaleY == null) return;

    Log.d(RENDER_TAG, "Applying scale: scaleX: $scaleX, scaleY: $scaleY")
    videoEffects += ScaleAndRotateTransformation.Builder()
        .setScale(scaleX ?: 1f, scaleY ?: 1f)
        .build()
}

File: macos/Classes/src/features/render/helpers/ApplyScale.swift
import CoreGraphics

public func applyScale(scaleX: Float?, scaleY: Float?) {
    let x = CGFloat(scaleX ?? 1.0)
    let y = CGFloat(scaleY ?? 1.0)

    VideoCompositor.scaleX = x
    VideoCompositor.scaleY = y

    if x != 1.0 || y != 1.0 {
        print("[\(Tags.render)] Applying scale: scaleX=\(x), scaleY=\(y)")
    }
}
